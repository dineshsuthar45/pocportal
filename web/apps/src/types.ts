import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
