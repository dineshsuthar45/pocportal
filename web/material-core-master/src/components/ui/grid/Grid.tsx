import { AgGridReact } from 'ag-grid-react'; // the AG Grid React Component
import { useEffect, useRef, useState } from 'react';

import { useTheme } from '@mui/material';
import { ColDef, ColGroupDef, IFilterOptionDef, SideBarDef } from 'ag-grid-community';
import 'ag-grid-community/styles/ag-grid.css';
import 'ag-grid-community/styles/ag-theme-balham.css';
import 'ag-grid-community/styles/agGridAlpineFont.css';
import { GridProps } from './Grid.types';
import './gridCommon.scss';
import './gridDark.scss';
import './gridLight.scss';

import classNames from 'classnames';
import { ISimpleFilterModelType } from 'ag-grid-community/dist/lib/filter/provided/simpleFilter';
import {
	blankFilter,
	defaultDateFilterOptions,
	defaultMultiFilters,
	defaultNumberFilterOptions,
	defaultTextFilterOptions,
	notBlankFilter,
} from './constants';

/**
 * Further properties can be found in AG-Grid website at https://www.ag-grid.com/react-data-grid/
 */

export const MGrid = <T extends any>({
	defaultColDef = { floatingFilter: true, sortable: true },
	skipInferNumberColumn = false,
	skipMaxTwoLineHeaderText = false,
	rowSelection = 'single',
	size = 'comfy',
	sideBar = {
		hiddenByDefault: false,
		position: 'right',
		toolPanels: [
			{
				id: 'columns',
				labelDefault: 'Columns',
				labelKey: 'columns',
				iconKey: 'columns',
				toolPanel: 'agColumnsToolPanel',
				minWidth: 225,
				maxWidth: 225,
				width: 225,
				toolPanelParams: {
					suppressPivots: true,
					suppressPivotMode: true,
				},
			},
		],
	},
	...props
}: GridProps<T>) => {
	const theme = useTheme();
	const gridRef = useRef<AgGridReact<T>>(null);
	const [themeMode, setThemeMode] = useState<'light' | 'dark'>(theme?.palette?.mode ?? 'light');
	const [rowData, setRowData] = useState<T[]>(props.rowData ?? []);
	const [columns, setColumns] = useState<(ColDef | ColGroupDef)[]>();
	const [defaultColumns, setDefaultColumns] = useState<ColDef>();

	const sideBarVisible =
		typeof sideBar === 'undefined'
			? false
			: sideBar === null
			? false
			: typeof sideBar === 'string'
			? sideBar.replace(/\s/g, '') !== ''
			: typeof sideBar === 'boolean'
			? sideBar
			: Array.isArray(sideBar)
			? sideBar.some((x) => x.replace(/\s/g, '') !== '')
			: typeof sideBar === 'object'
			? !(sideBar as SideBarDef).hiddenByDefault
			: false;

	useEffect(() => {
		setThemeMode(theme.palette.mode);
	}, [theme.palette.mode]);

	useEffect(() => {
		if (props.loadFromURL) {
			fetch(props.loadFromURL)
				.then((result) => result.json())
				.then((rowData) => setRowData(rowData));
		}
	}, [props.loadFromURL]);

	useEffect(() => {
		const isApiRefDefined = (gridRef: React.RefObject<AgGridReact<T>>) =>
			gridRef && gridRef.current && gridRef.current.api;
		const isColDef = (col: ColDef<T> | ColGroupDef<T>): col is ColDef<T> => 'field' in col || 'type' in col;

		const tranform = (columnDef: ColDef<T> | ColGroupDef<T>): ColDef<T> | ColGroupDef<T> => {
			if (isColDef(columnDef)) {
				const colDef = columnDef as ColDef<T>;
				const defaultMaxHeaderHeightCellClass = !skipMaxTwoLineHeaderText ? 'm-header-cell-text-default' : '';

				switch (typeof colDef.headerClass) {
					case 'string':
						colDef.headerClass += ` ${defaultMaxHeaderHeightCellClass}`;
						colDef.headerTooltip =
							!colDef.headerTooltip && colDef.headerName ? colDef.headerName : colDef.headerTooltip ?? colDef.field;
						break;
					case 'object':
						if (Array.isArray(colDef.headerClass)) {
							colDef.headerClass.concat(defaultMaxHeaderHeightCellClass);
							colDef.headerTooltip =
								!colDef.headerTooltip && colDef.headerName ? colDef.headerName : colDef.headerTooltip ?? colDef.field;
						}
						break;
					case 'function':
						break;
					default:
						colDef.headerClass = defaultMaxHeaderHeightCellClass;
						colDef.headerTooltip =
							!colDef.headerTooltip && colDef.headerName ? colDef.headerName : colDef.headerTooltip ?? colDef.field;
						break;
				}

				const getFilterParams = (filter: any, filterParams: any) => {
					const filterOptions = filterParams?.filterOptions as (IFilterOptionDef | ISimpleFilterModelType)[];

					if (filterOptions) {
						const indexNotBlank = filterOptions.findIndex((option) => option === 'notBlank');
						const indexBlank = filterOptions.findIndex((option) => option === 'blank');
						if (indexNotBlank > 0) filterParams.filterOptions[indexNotBlank] = notBlankFilter;
						if (indexBlank > 0) filterParams.filterOptions[indexBlank] = blankFilter;
					} else
						switch (filter) {
							// use default filter options for filter types which can contain 'blank'/'notBlank' filter
							case 'agNumberColumnFilter':
								return {
									...filterParams,
									filterOptions: defaultNumberFilterOptions,
								};
							case 'agTextColumnFilter':
								return {
									...filterParams,
									filterOptions: defaultTextFilterOptions,
								};
							case 'agDateColumnFilter':
								return {
									...filterParams,
									filterOptions: defaultDateFilterOptions,
								};
						}
					return filterParams;
				};
				const setMultiFilterParams = () => {
					const filterParams = colDef.filterParams;
					if (filterParams && filterParams.filters) {
						colDef.filterParams.filters?.forEach((value: any) => {
							value.filterParams = getFilterParams(value.filter, value.filterParams);
						});
					} else
						colDef.filterParams = {
							...filterParams,
							filters: defaultMultiFilters,
						};
				};

				if (colDef.filter)
					colDef.filter === 'agMultiColumnFilter'
						? setMultiFilterParams()
						: (colDef.filterParams = getFilterParams(colDef.filter, colDef.filterParams));

				return {
					...colDef,
					type:
						!skipInferNumberColumn && rowData.every((row) => typeof row[colDef.field as keyof typeof row] === 'number')
							? 'numericColumn'
							: colDef.type,
				};
			} else {
				const colGroupDef = columnDef as ColGroupDef<T>;
				const children = colGroupDef.children.map((child) => {
					return tranform(child);
				});

				const emptyHeaderClass = !colGroupDef.headerName ? 'ag-header-group-cell-with-group-empty' : '';

				switch (typeof colGroupDef.headerClass) {
					case 'string':
						colGroupDef.headerClass += ` ${emptyHeaderClass}`;
						break;
					case 'object':
						if (Array.isArray(colGroupDef.headerClass)) {
							colGroupDef.headerClass.concat(emptyHeaderClass);
						}
						break;
					case 'function':
						break;
					default:
						colGroupDef.headerClass = emptyHeaderClass;
						break;
				}

				return {
					...colGroupDef,
					children,
				};
			}
		};

		const newColumns = (isApiRefDefined(gridRef) ? gridRef.current?.api.getColumnDefs() : props.columnDefs) ?? [];
		newColumns.forEach((newColumn) => {
			Object.assign(newColumn, tranform(newColumn));
		});

		if (isApiRefDefined(gridRef)) {
			gridRef.current?.api.setColumnDefs(newColumns);
		}
		setColumns(newColumns);
	}, [rowData, props.columnDefs, gridRef, skipInferNumberColumn, skipMaxTwoLineHeaderText]);

	useEffect(() => {
		const newDefaultColumns = { ...defaultColDef };
		newDefaultColumns.autoHeaderHeight = skipMaxTwoLineHeaderText ? newDefaultColumns.autoHeaderHeight : true;
		newDefaultColumns.wrapHeaderText = skipMaxTwoLineHeaderText ? newDefaultColumns.wrapHeaderText : true;

		setDefaultColumns(newDefaultColumns);
	}, [defaultColDef, skipMaxTwoLineHeaderText]);

	useEffect(() => {
		gridRef?.current?.api?.resetRowHeights();
	}, [size]);
	const handleChangeRowHeight = () => (size === 'compact' ? 20 : 26);

	return (
		<div
			className={classNames(`m-grid ag-theme-balham ag-theme-material ag-theme-material-${themeMode} ${size}`, {
				'side-bar-visible': sideBarVisible,
			})}
			style={{ width: '100%', height: '100%' }}
		>
			<AgGridReact<T>
				ref={gridRef}
				rowData={rowData}
				defaultColDef={defaultColumns}
				columnDefs={columns}
				rowSelection={rowSelection}
				sideBar={sideBar}
				getRowHeight={handleChangeRowHeight}
				{...props}
			/>
		</div>
	);
};

MGrid.defaultProps = {
	defaultColDef: { floatingFilter: true, sortable: true },
	skipInferNumberColumn: false,
	skipMaxTwoLineHeaderText: false,
	rowSelection: 'single',
	sideBar: {
		hiddenByDefault: false,
		position: 'right',
		toolPanels: [
			{
				id: 'columns',
				labelDefault: 'Columns',
				labelKey: 'columns',
				iconKey: 'columns',
				toolPanel: 'agColumnsToolPanel',
				minWidth: 225,
				maxWidth: 225,
				width: 225,
				toolPanelParams: {
					suppressPivots: true,
					suppressPivotMode: true,
				},
			},
		],
	},
};
