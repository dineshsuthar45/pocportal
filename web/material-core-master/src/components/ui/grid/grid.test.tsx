import { render } from "@testing-library/react";
import React from "react";
import { MGrid } from "./Grid";
import { GridProps } from "./Grid.types";

describe("Grid component", () => {
  interface GripOption {
    athlete: string;
    age: number;
    country: string;
    sport: string;
    total: number;
    gold: number;
    silver: number;
    bronze: number;
  }
  const renderComponent = () => render(<MGrid<GripOption> {...props} />);
  let props: GridProps<GripOption>;

  beforeEach(() => {
    props = {};
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
