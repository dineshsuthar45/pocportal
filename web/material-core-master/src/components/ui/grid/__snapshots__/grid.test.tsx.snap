// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render correctly 1`] = `
<div>
  <div
    class="m-grid ag-theme-balham ag-theme-material ag-theme-material-light comfy side-bar-visible"
    style="width: 100%; height: 100%;"
  >
    <div
      style="height: 100%;"
    >
      <!-- AG Grid -->
      <div
        class="ag-root-wrapper ag-ltr ag-layout-normal"
        role="presentation"
      >
        <div
          class="ag-root-wrapper-body ag-focus-managed ag-layout-normal"
          role="presentation"
        >
          <div
            class="ag-tab-guard ag-tab-guard-top"
            role="presentation"
            tabindex="0"
          />
          <!-- AG Grid Body -->
          <div
            aria-colcount="0"
            aria-rowcount="1"
            class="ag-root ag-unselectable ag-layout-normal"
            role="grid"
          >
            <div
              class="ag-header ag-pivot-off"
              role="presentation"
              style="height: 33px; min-height: 33px;"
            >
              <div
                class="ag-pinned-left-header ag-hidden"
                role="presentation"
                style="width: 0px; min-width: 0px; max-width: 0px;"
              >
                <div
                  aria-rowindex="1"
                  class="ag-header-row ag-header-row-column"
                  role="row"
                  style="height: 32px; top: 0px; width: 0px;"
                />
              </div>
              <div
                class="ag-header-viewport "
                role="presentation"
              >
                <div
                  class="ag-header-container"
                  role="rowgroup"
                  style="width: 0px;"
                >
                  <div
                    aria-rowindex="1"
                    class="ag-header-row ag-header-row-column"
                    role="row"
                    style="height: 32px; top: 0px; width: 0px;"
                  />
                </div>
              </div>
              <div
                class="ag-pinned-right-header ag-hidden"
                role="presentation"
                style="width: 0px; min-width: 0px; max-width: 0px;"
              >
                <div
                  aria-rowindex="1"
                  class="ag-header-row ag-header-row-column"
                  role="row"
                  style="height: 32px; top: 0px; width: 0px;"
                />
              </div>
            </div>
            <!-- AG Pinned Top -->
            <div
              class="ag-floating-top"
              role="presentation"
              style="height: 0px; min-height: 0; display: none; overflow-y: hidden;"
            >
              <!-- AG Row Container topLeft -->
              <div
                class="ag-pinned-left-floating-top ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container topCenter -->
              <div
                class="ag-floating-top-viewport"
                role="presentation"
              >
                <div
                  class="ag-floating-top-container"
                  role="presentation"
                  style="width: 0px;"
                />
              </div>
              <!-- AG Row Container topRight -->
              <div
                class="ag-pinned-right-floating-top ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container topFullWidth -->
              <div
                class="ag-floating-top-full-width-container"
                role="presentation"
              />
            </div>
            <!-- AG Middle -->
            <div
              class="ag-body-viewport ag-row-no-animation ag-layout-normal"
              role="presentation"
            >
              <!-- AG Row Container left -->
              <div
                class="ag-pinned-left-cols-container ag-hidden"
                role="presentation"
                style="height: 1px;"
              />
              <!-- AG Row Container center -->
              <div
                class="ag-center-cols-clipper"
                role="presentation"
                style="height: 1px;"
              >
                <div
                  class="ag-center-cols-viewport"
                  role="presentation"
                  style="height: 100%;"
                >
                  <div
                    class="ag-center-cols-container"
                    role="presentation"
                    style="height: 1px; width: 0px;"
                  />
                </div>
              </div>
              <!-- AG Row Container right -->
              <div
                class="ag-pinned-right-cols-container ag-hidden"
                role="presentation"
                style="height: 1px;"
              />
              <!-- AG Row Container fullWidth -->
              <div
                class="ag-full-width-container"
                role="presentation"
                style="height: 1px;"
              />
            </div>
            <!-- AG Sticky Top -->
            <div
              class="ag-sticky-top"
              role="presentation"
              style="height: 0px; top: 33px; width: 100%;"
            >
              <!-- AG Row Container stickyTopLeft -->
              <div
                class="ag-pinned-left-sticky-top ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container stickyTopCenter -->
              <div
                class="ag-sticky-top-viewport"
                role="presentation"
              >
                <div
                  class="ag-sticky-top-container"
                  role="presentation"
                  style="width: 0px;"
                />
              </div>
              <!-- AG Row Container stickyTopRight -->
              <div
                class="ag-pinned-right-sticky-top ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container stickyTopFullWidth -->
              <div
                class="ag-sticky-top-full-width-container"
                role="presentation"
              />
            </div>
            <!-- AG Pinned Bottom -->
            <div
              class="ag-floating-bottom"
              role="presentation"
              style="height: 0px; min-height: 0; display: none; overflow-y: hidden;"
            >
              <!-- AG Row Container bottomLeft -->
              <div
                class="ag-pinned-left-floating-bottom ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container bottomCenter -->
              <div
                class="ag-floating-bottom-viewport"
                role="presentation"
              >
                <div
                  class="ag-floating-bottom-container"
                  role="presentation"
                  style="width: 0px;"
                />
              </div>
              <!-- AG Row Container bottomRight -->
              <div
                class="ag-pinned-right-floating-bottom ag-hidden"
                role="presentation"
              />
              <!-- AG Row Container bottomFullWidth -->
              <div
                class="ag-floating-bottom-full-width-container"
                role="presentation"
              />
            </div>
            <!-- AG Fake Horizontal Scroll -->
            <div
              aria-hidden="true"
              class="ag-body-horizontal-scroll ag-hidden"
              style="height: 0px; max-height: 0px; min-height: 0px;"
            >
              
            
              <div
                class="ag-horizontal-left-spacer"
                ref="eLeftSpacer"
                style="width: 0px; max-width: 0px; min-width: 0px;"
              />
              
            
              <div
                class="ag-body-horizontal-scroll-viewport"
                ref="eViewport"
                style="height: 0px; max-height: 0px; min-height: 0px;"
              >
                
                
                <div
                  class="ag-body-horizontal-scroll-container"
                  ref="eContainer"
                  style="height: 0px; max-height: 0px; min-height: 0px; width: 0px;"
                />
                
            
              </div>
              
            
              <div
                class="ag-horizontal-right-spacer"
                ref="eRightSpacer"
                style="width: 0px; max-width: 0px; min-width: 0px;"
              />
              
        
            </div>
            <!-- AG Overlay Wrapper -->
            <div
              aria-hidden="true"
              class="ag-overlay"
            >
              
            
              <div
                class="ag-overlay-panel"
              >
                
                
                <div
                  class="ag-overlay-wrapper ag-layout-normal ag-overlay-no-rows-wrapper"
                  ref="eOverlayWrapper"
                >
                  <span
                    class="ag-overlay-no-rows-center"
                  >
                    No Rows To Show
                  </span>
                </div>
                
            
              </div>
              
        
            </div>
          </div>
          <div
            class="ag-tab-guard ag-tab-guard-bottom"
            role="presentation"
            tabindex="0"
          />
        </div>
        <div
          class="ag-paging-panel ag-unselectable ag-hidden"
          id="ag-3"
        >
          
                
          <span
            class="ag-paging-row-summary-panel"
            role="status"
          >
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-3-first-row"
              ref="lbFirstRowOnPage"
            />
            
                    
            <span
              id="ag-3-to"
            >
              to
            </span>
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-3-last-row"
              ref="lbLastRowOnPage"
            />
            
                    
            <span
              id="ag-3-of"
            >
              of
            </span>
            
                    
            <span
              class="ag-paging-row-summary-panel-number"
              id="ag-3-row-count"
              ref="lbRecordCount"
            />
            
                
          </span>
          
                
          <span
            class="ag-paging-page-summary-panel"
            role="presentation"
          >
            
                    
            <div
              aria-label="First Page"
              class="ag-paging-button"
              ref="btFirst"
              role="button"
            >
              <span
                class="ag-icon ag-icon-first"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <div
              aria-label="Previous Page"
              class="ag-paging-button"
              ref="btPrevious"
              role="button"
            >
              <span
                class="ag-icon ag-icon-previous"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <span
              class="ag-paging-description"
              role="status"
            >
              
                        
              <span
                id="ag-3-start-page"
              >
                Page
              </span>
              
                        
              <span
                class="ag-paging-number"
                id="ag-3-start-page-number"
                ref="lbCurrent"
              />
              
                        
              <span
                id="ag-3-of-page"
              >
                of
              </span>
              
                        
              <span
                class="ag-paging-number"
                id="ag-3-of-page-number"
                ref="lbTotal"
              />
              
                    
            </span>
            
                    
            <div
              aria-label="Next Page"
              class="ag-paging-button"
              ref="btNext"
              role="button"
            >
              <span
                class="ag-icon ag-icon-next"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                    
            <div
              aria-label="Last Page"
              class="ag-paging-button"
              ref="btLast"
              role="button"
            >
              <span
                class="ag-icon ag-icon-last"
                role="presentation"
                unselectable="on"
              />
            </div>
            
                
          </span>
          
            
        </div>
      </div>
    </div>
  </div>
</div>
`;
